native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/config/AbstractMergeAgentFilesMojo.java

```diff
@@ -58,32 +58,24 @@ public abstract class AbstractMergeAgentFilesMojo extends AbstractMojo {
     @Component
     protected Logger logger;
 
-    protected File mergerExecutable;
+    private File mergerExecutable;
 
-    protected void tryInstallMergeExecutable(Path nativeImageExecutablePath) {
-        if (mergerExecutable != null && mergerExecutable.exists()) {
-            return;
+    public File getMergerExecutable() throws MojoExecutionException {
+        if (mergerExecutable == null) {
+            initializeMergerExecutable();
         }
 
-        File nativeImageExecutable = nativeImageExecutablePath.toAbsolutePath().toFile();
+        return mergerExecutable;
+    }
+
+    private void initializeMergerExecutable() throws MojoExecutionException {
+        Path nativeImage = NativeImageConfigurationUtils.getNativeImage(logger);
+        File nativeImageExecutable = nativeImage.toAbsolutePath().toFile();
         File mergerExecutable = new File(nativeImageExecutable.getParentFile(), nativeImageConfigureFileName());
         if (!mergerExecutable.exists()) {
-            getLog().info("Installing native image merger to " + mergerExecutable);
-            ProcessBuilder processBuilder = new ProcessBuilder(nativeImageExecutable.toString());
-            processBuilder.command().add("--macro:native-image-configure-launcher");
-            processBuilder.directory(mergerExecutable.getParentFile());
-            processBuilder.inheritIO();
-
-            try {
-                Process installProcess = processBuilder.start();
-                if (installProcess.waitFor() != 0) {
-                    getLog().warn("Installation of native image merging tool failed");
-                }
-                NativeImageUtils.maybeCreateConfigureUtilSymlink(mergerExecutable, nativeImageExecutablePath);
-            } catch (IOException | InterruptedException e) {
-                // ignore since we will handle that if the installer doesn't exist later
-            }
-
+            throw new MojoExecutionException("'" + nativeImageConfigureFileName() + "' tool was not found in your " + nativeImage + "." +
+                    "This probably means that the JDK at '" + nativeImage + "' is not a GraalVM distribution."
```

Comment:
`getNativeImage()` already checks whether `native-image` exists, so if `native-image-configure` is missing, then it is an incomplete GraalVM distribution. Maybe something like this:

```suggestion
            throw new MojoExecutionException("'" + nativeImageConfigureFileName() + "' tool was not found in the GraalVM JDK at '" + nativeImageExecutable.getParentFile().getParentFile() + "'." +
                    "This probably means that you are using a GraalVM distribution that is not fully supported by the Native Build Tools."
```